app/models/metadata.rb

```diff
@@ -51,65 +78,68 @@ def self.filters_to_json
         options: ["Marine spatial planning", "Education", "Ecosystem assessment", "Environmental impact assessment", "Ecosystem services"],
         sortButtons: false,
         type: "multiple"
-      },
-    ].to_json
-  end
-
-  def self.table_headers_to_json
-    table_headers = [
-      {
-        name: "category",
-        title: "Category",
-        sortButtons: true,
-        onMobile: false
-      },
-      {
-        name: "resource",
-        title: "Resource",
-        sortButtons: true,
-        onMobile: true
-      },
-      {
-        name: "contact_organisation",
-        title: "Contact Organisation",
-        sortButtons: true,
-        onMobile: false
-      },
-      {
-        name: "themes",
-        title: "Themes",
-        sortButtons: false,
-        onMobile: false
       }
     ].to_json
   end
 
-  def self.metadata_to_json
+  def self.metadata(params)
     output = []
-    metadata = Metadata.all.order(id: :asc)
+    filters = sanitize_params(params['filters'])
+    filter_data = params.dig('filters').present? ? Metadata.where(filters) : Metadata.all
+    metadata = sorting_data(params, filter_data)
     metadata.to_a.each do |meta|
       meta_attributes = meta.attributes
       meta_attributes[:metadata] = metadata_url(meta)
       meta_attributes[:themes] = []
       ["marine_spatial_planning", "education", "environmental_impact_assessment",
       "ecosystem_assessment", "ecosystem_services"].each do |attribute|
-        meta_attributes[:themes] << attribute.capitalize.gsub('_', ' ') if meta_attributes[attribute]
+        meta_attributes[:themes] << attribute.capitalize.tr('_', ' ') if meta_attributes[attribute]
         meta_attributes.delete(attribute)
       end
-      meta_attributes.delete_if { |k, v| ["created_at", "updated_at"].include? k }
+      meta_attributes.delete_if { |k| ["created_at", "updated_at"].include? k }
       output << meta_attributes
     end
-    output.to_json
+    pagination(params['requested_page'], output, filter_data)
+  end
+
+  def self.pagination(page, items, total)
```

Comment:
Can we please move this into a serializer?
I believe this is getting passed to the frontend so I think it would make sense to move all data formatting there where possible